core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -0,0 +1,270 @@
+    every { propertyQuery.findStrings() } returns bookmarkUrlArray
+    assertThat(newBookmarksDao.getCurrentZimBookmarksUrl(zimFileReader))
+      .isEqualTo(listOf(bookmarkUrl))
+  }
+
+  @Test
+  fun `getCurrentZimBookmarksUrl returns list of distinct bookmarkUrl`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    val zimId = ""
+    every { zimFileReader?.id } returns zimId
+    every { queryBuilder.equal(BookmarkEntity_.zimId, zimFileReader?.id) } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    val zimName = ""
+    every { zimFileReader?.name } returns zimName
+    every { queryBuilder.equal(BookmarkEntity_.zimName, zimFileReader?.name) } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkUrl = "bookmarkUrl"
+    val propertyQuery: PropertyQuery = mockk()
+    every { query.property(BookmarkEntity_.bookmarkUrl) } returns propertyQuery
+    val bookmarkUrlArray = arrayOf<String>(bookmarkUrl, bookmarkUrl)
+    every { propertyQuery.findStrings() } returns bookmarkUrlArray
+    assertThat(newBookmarksDao.getCurrentZimBookmarksUrl(zimFileReader))
+      .isEqualTo(listOf(bookmarkUrl))
+  }
+
+  @Test
+  fun `getCurrentZimBookmarksUrl returns empty list with empty zimName and zimId`() {
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.equal(BookmarkEntity_.zimName, "") } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    every { queryBuilder.equal(BookmarkEntity_.zimId, "") } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    assertThat(newBookmarksDao.getCurrentZimBookmarksUrl(null)).isEmpty()
+  }
+
+  @Test
+  fun `bookmark urls for current workbook`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { zimFileReader.name } returns ""
+    every { queryBuilder.equal(BookmarkEntity_.zimName, zimFileReader.name) } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    every { zimFileReader.id } returns ""
+    every { queryBuilder.equal(BookmarkEntity_.zimId, zimFileReader.id) } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    newBookmarksDao.bookmarkUrlsForCurrentBook(zimFileReader).test()
+      .assertEmpty()
```

Comment:
I want to see more than `assertEmpty` you can look at RecentSearchDaoTest

```
      val queryBuilder = mockk<QueryBuilder<RecentSearchEntity>>()
      every { box.query() } returns queryBuilder
      every { queryBuilder.equal(RecentSearchEntity_.zimId, zimId) } returns queryBuilder
      every { queryBuilder.orderDesc(RecentSearchEntity_.id) } returns queryBuilder
      val query = mockk<Query<RecentSearchEntity>>()
      every { queryBuilder.build() } returns query
      mockkStatic(RxQuery::class)
      every { RxQuery.observable(query) } returns Observable.just(queryResult)
    }
```

for an example of how you deal with `box.asFlowable`